### ISSUE TITLE
Add a challenge mode where you only battle

### ISSUE THREAD

**Issue by spollard** [Tuxemon/Tuxemon#2316]
**Is your feature request related to a problem? Please describe.**
In a way, yes. The main game seems mostly playable, though last time I tried I ran into enough problems to stop playing the Cathedral campaign. 

**Describe the solution you'd like**
It seemed like most of my issues were with the map interactions but the battles worked well, so I was imagining creating a challenge mode which is a series of fights where you have a set of tuxemon and you battle different trainers. You can inspect/read about the enemy tuxemon (levels, moves, etc) before choosing the challenge. Then you don't care if the rest of the game (map, evolutions, all the moves, all the tuxemon, etc) are finished or working.

The benefits of doing this: You can present a complete game that is fun to play, but small enough to be polished. Some people would enjoy the puzzle of winning battles! It would depend on way fewer features working. All the interaction with your character on the map (evolution, fishing rod, walking around, catching new 'mons, etc) wouldn't need to work. Yes, I know that walking around the map is part of the fun and storyline! 

What would this entail: adding a challenge mode to the main menu (I already did this, it was easy). Make a menu screen for selecting which challenge to attempt. Make a screen so you can inspect the challenge (view your tuxemons, view their tuxemons) before attempting it. After selecting a challenge, start the battle with the specific monsters that you have and possibly items. Have a victory screen or lost screen that will kick you back to the select a challenge screen. If you win the challenge, unlock the next challenge. I think 20 challenges would be enough to be satisfying, but we could start with like 5 of course. Also, we could simply reuse encounters with trainers that are already in the game. 

**Describe alternatives you've considered**
Possible extensions: turn the linear challenge into a branching challenge tree where players can choose which challenge to attempt next. Perhaps you may select four to six monsters to start with, then they all will gain some amount of experience after each battle and your goal is to defeat the final boss/challenge (almost like elite 4 style). There may be perhaps four or eight different battles you'd have to face. Perhaps add terrain types in the branches of the trees that require you to have a specific kind of monster in order to move across it, such as you must have a water monster in order to move across water.

**Additional context**
Code that may be helpful

Random encounter might be easier to start with
tuxemon/event/actions/random_encounter.py:24:    Randomly starts a battle with a monster defined in the "encounter" table

Random battle with npc
tuxemon/event/actions/random_battle.py:26:    Start random battle with a random npc with a determined
I tried to start this action from the cli, but I haven't figured out how to create or load a world state yet....

start battle
tuxemon/event/actions/start_battle.py:22:    Start a battle between two characters and switch to the combat module.

Start a trainer battle from command line!
tuxemon/cli/commands/trainer_battle.py:15:class TrainerBattleCommand(CLICommand):
I thought this would be helpful, but I don't know how to invoke it...

**Comment by Davidliang11292** [comment]
great idea! After we had the No battle mode, we had the Challenge mode, which gave players more options

**Comment by spollard** [comment]
I see this idea was floated before in this comment, though it was on an unrelated issue. https://github.com/Tuxemon/Tuxemon/issues/509#issuecomment-460932072

**Comment by Sanglorian** [comment]
Great idea!

**Comment by HippasusTwo** [comment]
One possibility is a 'campaign' that is actually just pretty much getting a tuxemon team and doing battles. Call it Tuxemon Stadium.

So it would consist of a large single map that auto-heals you after every battle, a tuxemon team swapper person/pc, a 'random wild battle trigger' person, a specific scenario trigger person, a random trainer person, and an 'ask to fight specific monster/trainer' person(s). It could store victories and losses on a sign board or something. You'd mainly just need the ability to input a tuxemon number / trainer ID, which I think can already be done.

**Comment by Sanglorian** [comment]
> One possibility is a 'campaign' that is actually just pretty much getting a tuxemon team and doing battles. Call it Tuxemon Stadium.
> 
> So it would consist of a large single map that auto-heals you after every battle, a tuxemon team swapper person/pc, a 'random wild battle trigger' person, a specific scenario trigger person, a random trainer person, and an 'ask to fight specific monster/trainer' person(s). It could store victories and losses on a sign board or something. You'd mainly just need the ability to input a tuxemon number / trainer ID, which I think can already be done.

I like this idea!

**Comment by JaskRendix** [comment]
@spollard back then you complained about the lack of animation when evolving, now it's implemented, you can find it in the link below, I hope it matches your standard, if not feel free to open another issue to address something you don't like as you have done before

https://github.com/Tuxemon/Tuxemon/pull/2390#issuecomment-2326380396

**Comment by spollard** [comment]
Ah you meant to at @HippasusTwo about the evolution stuff

**Comment by JaskRendix** [comment]
@spollard nope, you noticed it in #2035 I have shared that comment so you can evaluate the animation (video)

**Comment by HippasusTwo** [comment]
I've coded a mode/campaign that does about half of this, but it's currently lacking the ability to select any tuxemon you want, and also has the additional feature that it has a 'game over' ability if you get a full team knock out. There are optional sidequests and a map to roam, or you can just go in one big building, talk to a nurse and get a full team, and start fighting random trainers (their strength needs a bit of tweaking to scale properly with the trainer I think, but there is some scaling). You can track how many wins you've had in a current run without a 'death'.

I was making it with a different kind of challenge in mind, based on the old 'Nuzlocke challenge' for pokemon where full team knockout means 'death' for the player and you get a lot of randomization in your team and few critters, but there's a lot of overlap between that and this, and what one could do is just let one select what challenge exactly one is going for: unlimited mon selection at max levels and just pure random fights, or trying to survive with a limited team? I had a storyline in mind, the Democratic Kingdom of Vivipere getting attacked by people who want to overthrow its democracy, but the story part is completely optional.

I need to go find the part of the code that apparently hardcodes what campaigns are 'expected', it is working but the style checker is yelling at me that my campaign is illegally named or something lol. edit: found it, test folder maps tmx and yaml were the culprits.
